src/serialization/Portable.ts

```diff
@@ -41,127 +48,552 @@ export enum FieldType {
+     *
+     * @param fieldName name of the field
+     * @param value     int value to be written
+     */
     writeShort(fieldName: string, value: number): void;
 
-    writePortable(fieldName: string, portable: Portable): void;
+    /**
+     * Writes a Portable.
+     * Use {@link writeNullPortable} to write a `null` Portable
+     *
+     * @param fieldName name of the field
+     * @param portable  Portable to be written
+     */
+    writePortable(fieldName: string, portable: Portable | null): void;
 
+    /**
+     * To write a null portable value, user needs to provide class and factoryIds of related class.
+     *
+     * @param fieldName name of the field
+     * @param factoryId factory ID of related portable class
+     * @param classId   class ID of related portable class
+     */
     writeNullPortable(fieldName: string, factoryId: number, classId: number): void;
 
-    writeByteArray(fieldName: string, bytes: Buffer): void;
+    /**
+     * Writes a decimal which is arbitrary precision and scale floating-point number
+     *
+     * @param fieldName name of the field
+     * @param value     BigDecimal value to be written
+     */
+    writeDecimal(fieldName: string, value: BigDecimal | null): void;
 
-    writeBooleanArray(fieldName: string, booleans: boolean[]): void;
+    /**
+     * Write a time field consisting of hour, minute, seconds and nanos parts
+     *
+     * @param fieldName name of the field
+     * @param value     LocalTime value to be written
+     */
+    writeTime(fieldName: string, value: LocalTime | null): void
 
-    writeCharArray(fieldName: string, chars: string[]): void;
+    /**
+     * Writes a date field consisting of year, month of the year and day of the month
+     *
+     * @param fieldName name of the field
+     * @param value     LocalDate value to be written
+     */
+    writeDate(fieldName: string, value: LocalDate | null): void;
 
-    writeIntArray(fieldName: string, ints: number[]): void;
+    /**
+     * Writes a timestamp field consisting of
+     * year, month of the year, day of the month, hour, minute, seconds, nanos parts
+     *
+     * @param fieldName name of the field
+     * @param value     LocalDateTime value to be written
+     */
+    writeTimestamp(fieldName: string, value: LocalDateTime | null): void;
 
-    writeLongArray(fieldName: string, longs: Long[]): void;
+    /**
+     * Writes a timestamp with timezone field consisting of
+     * year, month of the year, day of the month, offset seconds, hour, minute, seconds, nanos parts
+     *
+     * @param fieldName name of the field
+     * @param value     OffsetDateTime value to be written
+     */
+    writeTimestampWithTimezone(fieldName: string, value: OffsetDateTime | null): void;
 
-    writeDoubleArray(fieldName: string, doubles: number[]): void;
+    /**
+     * Writes a primitive byte-array.
+     *
+     * @param fieldName name of the field
+     * @param bytes     byte array to be written
```

Comment:
`byte array` or `byte-array`? I think it would be good to pick one style and apply it to everywhere. Similar comment for others. Personally, I would prefer `byte array` but I am OK with other also